src/enclave/curl.h

```diff
@@ -89,25 +91,23 @@ namespace afetch {
       curl_easy_setopt(curl, CURLOPT_CAPATH, NULL);
 
       auto res = curl_easy_perform(curl);
-
       if (res != CURLE_OK) {
-        const char* err_s = curl_easy_strerror(res);
-        throw CurlOtherError(url, err_s, res);
+        response.error_message = curl_easy_strerror(res);
+        if (verbose) {
+          std::cerr << "Fetch failed: " << response.error_message << std::endl;
+        }
+        response.status = NULL;
+        return response;
       }
 
-      int64_t response_code;
-      curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &response_code);
+      curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &response.status);
 
       if (verbose) {
         std::cerr << "URL: " << url << std::endl;
-        std::cerr << "Response code:" << response_code << std::endl;
+        std::cerr << "Response status code: " << response.status << std::endl;
         std::cerr << "Response body:" << std::string((char*)response.body.data(), response.body.size()) << std::endl;
       }
 
-      if (response_code >= 400) {
-        throw CurlHTTPError(url, response_code);
```

Comment:
The `CurlHTTPError` class defined above can be removed now, since it isn't used anywhere.